src/Components/MagnifyImage.tsx

```diff
@@ -63,9 +55,8 @@ export const MagnifyImage: React.FC<React.PropsWithChildren<MagnifyImageProps>>
     const x = e.clientX - containerRect.left
     const y = e.clientY - containerRect.top
 
-    setPositions({
-      x,
-      y,
+    requestAnimationFrame(() => {
```

Comment:
☝️ the fix

Reply:
Nice! So I guess without `requestAnimationFrame`, it'd be rendering faster than it needs to be and that caused the possible janky animation.